doc/pitfalls/data_leakage.rst

```diff
@@ -0,0 +1,152 @@
+:class:`~sklearn.impute.SimpleImputer` allows you to replace missing values
+with the mean of that feature. Only the train data should be used to
+calculate this mean value as including the test data in the mean calculation
+will introduce information about the test data into the model.
+
+To demonstrate this, we will use the :ref:`diabetes_dataset` and
+artificially introduce missing values. We introduce a large number of
+missing values to make the effect more pronounced::
+
+    >>> import numpy as np
+    >>> from sklearn.datasets import load_diabetes
+    >>> X, y = load_diabetes(return_X_y=True)
+    >>> rng = np.random.RandomState(42)
+    >>> n_samples = X.shape[0]
+    >>> n_features = X.shape[1]
+    >>> n_missing = int(n_samples * 0.75)
+    >>> missing_samples = np.zeros(n_samples, dtype=np.bool)
+    >>> missing_samples[: n_missing] = True
+    >>> rng.shuffle(missing_samples)
+    >>> missing_features = rng.randint(0, n_features, n_missing)
+    >>> X_missing = X.copy()
+    >>> X_missing[missing_samples, missing_features] = np.nan
+
+**Wrong**
+
+Using all the data to calculate the feature means, to replace the missing
+values with, results in a very high accuracy::
+
+    >>> from sklearn.impute import SimpleImputer
+    >>> from sklearn.ensemble import GradientBoostingRegressor
+    >>> from sklearn.model_selection import cross_val_score
+    >>> X_impute = SimpleImputer().fit_transform(X_missing)
+    >>> scores = cross_val_score(GradientBoostingRegressor(random_state=1),
+    ...                          X_impute, y)
+    >>> print(f"Mean R^2: {scores.mean():.3f}+/-{scores.std():.2f}")
+    Mean R^2: 0.412+/-0.06
+
+**Right**
+
+Using a :class:`~sklearn.pipeline.Pipeline` to chain together the imputation
+and model ensures that only the train data subset is using for imputation.
+This results in a much lower accuracy::
+
+    >>> from sklearn.pipeline import make_pipeline
+    >>> pipeline = make_pipeline(SimpleImputer(),
+    ...                          GradientBoostingRegressor(random_state=1))
+    >>> scores = cross_val_score(pipeline, X_missing, y)
+    >>> print(f"Mean R^2: {scores.mean():.3f}+/-{scores.std():.2f}")
+    Mean R^2: 0.421+/-0.07
+
+Use pipelines
+=============
+
+You may have noticed a common theme in our examples. Both the 'Right' examples
+use the :ref:`pipeline <pipeline>`, which helps prevent data leakage by
+only using the training data to calculate preprocessing statistics. Conversely,
+both the 'Wrong' examples used the :term:`fit_transform` method of the
+preprocessing estimator.
+Care needs to be taken when using the :term:`fit_transform` method. This is
+because it combines the `fit` method, which should
+only be performed on the train subset, and the :term:`transform` method which
+is generally performed on the whole dataset, as the train and test subsets
+should be preprocessed in the same way. Scikit-learn pipelines ensure that
+the appropriate method is performed on the correct data subset.
```

Comment:
You can basically saying that fit_transform(X) is bad
but I would also highlight that fit_transform(X_test) is also wrong 
as it can harm performance